server/src/main/java/io/druid/server/coordination/ZkCoordinator.java

```diff
@@ -256,8 +256,14 @@ public void loadLocalCache()
       File file = segmentsToLoad[i];
       log.info("Loading segment cache file [%d/%d][%s].", i, segmentsToLoad.length, file);
       try {
-        DataSegment segment = jsonMapper.readValue(file, DataSegment.class);
-        if (serverManager.isSegmentCached(segment)) {
+        final DataSegment segment = jsonMapper.readValue(file, DataSegment.class);
+
+        if (!segment.getIdentifier().equals(file.getName())) {
```

Comment:
My thinking behind keeping it around is that we're _not_ deleting the actual segment data files (mostly because the deletion code doesn't work if the dataSegment object doesn't match the path) and so we might as well keep the cache file around too as a pointer.